aws-sdk/aws-sdk.d.ts

```diff
@@ -382,7 +382,7 @@ declare module "aws-sdk" {
     endpoint: Endpoint;
 
     getObject(params: s3.GetObjectRequest, callback?: (err: Error, data: any) => void): any;
-    putObject(params: s3.PutObjectRequest, callback: (err: Error, data: any) => void): void;
+    putObject(params: s3.PutObjectRequest, callback?: (err: Error, data: any) => void): any;
```

Comment:
I like overload, but was trying to be consistent with s3.GetObjectRequest. I suggest another PR that refactors out the use of any return type just as you have described.